plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/flex/CreateAppEngineFlexWtpProject.java

```diff
@@ -16,28 +16,76 @@
 import com.google.cloud.tools.eclipse.appengine.newproject.CodeTemplates;
 import com.google.cloud.tools.eclipse.appengine.newproject.CreateAppEngineWtpProject;
 import com.google.cloud.tools.eclipse.appengine.newproject.Messages;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.apache.maven.artifact.Artifact;
 import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IFolder;
 import org.eclipse.core.resources.IProject;
 import org.eclipse.core.runtime.CoreException;
 import org.eclipse.core.runtime.IAdaptable;
+import org.eclipse.core.runtime.IPath;
 import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.core.runtime.SubMonitor;
+import org.eclipse.jdt.core.IClasspathEntry;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jst.common.project.facet.core.JavaFacet;
+import org.eclipse.jst.j2ee.web.project.facet.WebFacetUtils;
+import org.eclipse.wst.common.project.facet.core.IFacetedProject;
+import org.eclipse.wst.common.project.facet.core.ProjectFacetsManager;
 
 /**
-* Utility to make a new Eclipse project with the App Engine Flexible facets in the workspace.
-*/
+ * Utility to create a new App Engine Flexible Eclipse project.
+ */
 public class CreateAppEngineFlexWtpProject extends CreateAppEngineWtpProject {
+  private static final Logger logger = Logger.getLogger(CreateAppEngineFlexWtpProject.class.getName());
+  private static final Map<String, String> PROJECT_DEPENDENCIES;
+
+  static {
+    Map<String, String> projectDependencies = new HashMap<String, String>();
+    projectDependencies.put("javax.servlet", "servlet-api");
+    PROJECT_DEPENDENCIES = Collections.unmodifiableMap(projectDependencies);
+  }
 
-  CreateAppEngineFlexWtpProject(AppEngineProjectConfig config, IAdaptable uiInfoAdapter) {
+  private ILibraryRepositoryService repositoryService;
+
+  CreateAppEngineFlexWtpProject(AppEngineProjectConfig config, IAdaptable uiInfoAdapter,
+      ILibraryRepositoryService repositoryService) {
     super(config, uiInfoAdapter);
+    this.repositoryService = repositoryService;
   }
 
   @Override
   public void addAppEngineFacet(IProject newProject, IProgressMonitor monitor) throws CoreException {
-    // TODO add flex facet
+    SubMonitor subMonitor = SubMonitor.convert(monitor,
+        Messages.getString("add.appengine.flex.facet"), 100);
+
+    IFacetedProject facetedProject = ProjectFacetsManager.create(
+        newProject, true /* convertIfNecessary */, subMonitor.newChild(50));
+    AppEngineFlexFacet.installAppEngineFacet(
```

Comment:
It seems odd to have this call alongside `configureFacets()`. Each call to `IFacetedProject.modify()` triggers the Dependency Graph job, and more.  Perhaps this could take the install `Set<>`?